flutter/lib/src/frames_tracking/sentry_delayed_frames_tracker.dart

```diff
@@ -0,0 +1,303 @@
+  void resume() {
+    _isTrackingActive = true;
+  }
+
+  /// Pauses the collecting of frames.
+  void pause() {
+    _isTrackingActive = false;
+    _resetCurrentFrame();
+  }
+
+  /// Retrieves the frames the intersect with the provided [startTimestamp] and [endTimestamp].
+  @visibleForTesting
+  List<SentryFrameTiming> getFramesIntersecting(
+      {required DateTime startTimestamp, required DateTime endTimestamp}) {
+    return _delayedFrames.where((frame) {
+      // Fully contained or exactly matching
+      final fullyContainedOrMatching =
+          frame.startTimestamp.compareTo(startTimestamp) >= 0 &&
+              frame.endTimestamp.compareTo(endTimestamp) <= 0;
+
+      // Partially contained, starts before range, ends within range
+      final startsBeforeEndsWithin =
+          frame.startTimestamp.isBefore(startTimestamp) &&
+              frame.endTimestamp.isAfter(startTimestamp) &&
+              frame.endTimestamp.isBefore(endTimestamp);
+
+      // Partially contained, starts within range, ends after range
+      final startsWithinEndsAfter =
+          frame.startTimestamp.isAfter(startTimestamp) &&
+              frame.startTimestamp.isBefore(endTimestamp) &&
+              frame.endTimestamp.isAfter(endTimestamp);
+
+      return fullyContainedOrMatching ||
+          startsBeforeEndsWithin ||
+          startsWithinEndsAfter;
+    }).toList(growable: false);
+  }
+
+  /// Calculates the frame metrics based on start, end timestamps and the
+  /// delayed frames metrics. If the delayed frames array is empty then
+  /// only the total frames will be calculated.
+  SpanFrameMetrics? getFrameMetrics(
+      {required DateTime spanStartTimestamp,
+      required DateTime spanEndTimestamp}) {
+    final relevantFrames = getFramesIntersecting(
+        startTimestamp: spanStartTimestamp, endTimestamp: spanEndTimestamp);
+    final spanDuration =
+        spanEndTimestamp.difference(spanStartTimestamp).inMilliseconds;
+
+    // No slow or frozen frames detected
+    if (relevantFrames.isEmpty) {
+      return SpanFrameMetrics(
+          totalFrameCount:
+              (spanDuration / _expectedFrameDuration.inMilliseconds).ceil(),
+          slowFrameCount: 0,
+          frozenFrameCount: 0,
+          framesDelay: 0);
+    }
+
+    final spanStartMs = spanStartTimestamp.millisecondsSinceEpoch;
+    final spanEndMs = spanEndTimestamp.millisecondsSinceEpoch;
+    final expectedDurationMs = _expectedFrameDuration.inMilliseconds;
+    final frozenThresholdMs = _frozenFrameThreshold.inMilliseconds;
+
+    int slowFrameCount = 0;
+    int frozenFrameCount = 0;
+    int slowFramesDuration = 0;
+    int frozenFramesDuration = 0;
+    int framesDelay = 0;
+
+    for (final timing in delayedFrames) {
+      final frameStartMs = timing.startTimestamp.millisecondsSinceEpoch;
+      final frameEndMs = timing.endTimestamp.millisecondsSinceEpoch;
+      final actualDurationMs = timing.duration.inMilliseconds;
```

Comment:
Nitpick: Could call this `frameDurationMs`. It's easier to read (for me), as it relates to `frameStarMs` and `frameEndMs`